test/cql-pytest/test_permissions.py

```diff
@@ -553,3 +554,174 @@ def test_native_functions_always_exeutable(cql):
                     assert list(user_session.execute(f"SELECT max(a) FROM {table}")) == [(84,)]
                     assert list(user_session.execute(f"SELECT min(a) FROM {table}")) == [(3,)]
                     assert list(user_session.execute(f"SELECT sum(a) FROM {table}")) == [(102,)]
+
+# When a role has permissions to create a table, the role is given full
+# access to the table it just created (we call this feature "auto-grant").
+# The following tests check the auto-grant feature. The first test checks
+# that the base table created by CREATE TABLE is usable by the role creating
+# it. The second test is about the view created by CREATE MATERIALIZED VIEW
+# being accessible. The third test is about newly created CDC being accessible.
+def test_auto_grant_base(cql, test_keyspace):
+    with new_user(cql) as username:
+        # Grant username CREATE permissions on test_keyspace, but no other
+        # permissions:
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, username)
+        with new_session(cql, username) as user_session:
+            schema = "p int primary key"
+            with new_test_table(user_session, test_keyspace, schema) as table:
+                # Check that the username was auto-granted SELECT permissions
+                user_session.execute(f'SELECT * FROM {table}')
```

Comment:
I don't understand why auto-granted permissions not being propagated quickly enough were a problem for `DROP TABLE` but not for other statements like this one.

Reply:
To be honest, I may not have paid enough attention to the issue of retries in this test (unlike the separate tests I was writing in parallel for Alternator), but I did run the tests with long caching to see if I missed any place and supposedly I didn't. I'll look at it again.
Maybe I should use the eventually_* functions more or even everywhere in this test. I'll give it another thought.